bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/communication/RotelSource.java

```diff
@@ -399,57 +408,40 @@ public String getLabel() {
     }
 
     /**
-     * Get the command to select the source in the main zone
+     * Get the command to select the source in a zone
      *
-     * @return the command
-     */
-    public @Nullable RotelCommand getMainZoneCommand() {
-        return mainZoneCommand;
-    }
-
-    /**
-     * Get the command to select the source in the zone 2
+     * @param numZone the zone number, 1 for main zone or zone 1, 2 for zone 2, 3 for zone 3, 4 for zone 4
      *
      * @return the command
      */
-    public @Nullable RotelCommand getZone2Command() {
-        return zone2Command;
-    }
-
-    /**
-     * Get the command to select the source in the zone 3
-     *
-     * @return the command
-     */
-    public @Nullable RotelCommand getZone3Command() {
-        return zone3Command;
-    }
-
-    /**
-     * Get the command to select the source in the zone 4
-     *
-     * @return the command
-     */
-    public @Nullable RotelCommand getZone4Command() {
-        return zone4Command;
+    public @Nullable RotelCommand getZoneCommand(int numZone) {
+        switch (numZone) {
+            case 2:
+                return zone2Command;
+            case 3:
+                return zone3Command;
+            case 4:
+                return zone4Command;
+            default:
+                return zone1Command;
+        }
```

Comment:
Consider adding "case 1" for zone1Command and letting `default` throw `ArgumentException`.